src/ILCompiler.ReadyToRun/src/ObjectWriter/Relocator.cs

```diff
@@ -0,0 +1,559 @@
+    /// with RVA and delta parameters.
+    /// </summary>
+    public abstract class Relocator
+    {
+        /// <summary>
+        /// Retrieve the relocator for a given relocation type.
+        /// </summary>
+        /// <param name="relocationType">Relocation type</param>
+        /// <returns></returns>
+        public static Relocator GetRelocator(ushort relocationType)
+        {
+            switch (relocationType)
+            {
+                case RelocType.IMAGE_REL_BASED_ABSOLUTE:
+                case RelocType.IMAGE_REL_BASED_HIGHLOW:
+                    // TODO: ADDR32NB doesn't create a file-level relocation record while
+                    // the other three types do.
+                case RelocType.IMAGE_REL_BASED_ADDR32NB:
+                    return Absolute32.Instance;
+
+                case RelocType.IMAGE_REL_BASED_REL32:
+                    return Relative32.Instance;
+
+                case RelocType.IMAGE_REL_BASED_DIR64:
+                    // TODO: file-level relocation record
+                    return Absolute64.Instance;
+
+                case RelocType.IMAGE_REL_BASED_THUMB_MOV32:
+                    return ThumbMov32.Instance;
+
+                case RelocType.IMAGE_REL_BASED_THUMB_BRANCH24:
+                    return ThumbBranch24.Instance;
+/*
+    
+                case RelocType.IMAGE_REL_BASED_ARM64_BRANCH26:
+                case RelocType.IMAGE_REL_BASED_RELPTR32:
+                case RelocType.IMAGE_REL_SECREL:
+
+                case RelocType.IMAGE_REL_BASED_ARM64_PAGEBASE_REL21:
+                case RelocType.IMAGE_REL_BASED_ARM64_PAGEOFFSET_12A:
+                case RelocType.IMAGE_REL_BASED_ARM64_PAGEOFFSET_12L:
+*/
+
+                default:
+                    throw new NotSupportedException();
+            }
+        }
+
+        /// <summary>
+        /// Return file relocation type for the given relocation type. If the relocation
+        /// doesn't require a file-level relocation entry in the .reloc section, 0 is returned
+        /// corresponding to the IMAGE_REL_BASED_ABSOLUTE no-op relocation record.
+        /// </summary>
+        /// <param name="relocationType">Relocation type</param>
+        /// <returns>File-level relocation type or 0 (IMAGE_REL_BASED_ABSOLUTE) if none is required</returns>
+        public static ushort GetFileRelocationType(ushort relocationType)
+        {
+            switch (relocationType)
+            {
+                case RelocType.IMAGE_REL_BASED_HIGHLOW:
+                case RelocType.IMAGE_REL_BASED_DIR64:
+                case RelocType.IMAGE_REL_BASED_THUMB_MOV32:
+                    return relocationType;
+                    
+                default:
+                    return RelocType.IMAGE_REL_BASED_ABSOLUTE;
+            }
+        }
+
+        /// <summary>
+        /// Number of bytes this relocator needs to process
+        /// </summary>
+        /// <returns></returns>
+        abstract public int GetLength();
```

Comment:
Nit: We try to follow corefx coding style - visibility first. https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md

Reply:
Thanks for the heads-up, fixed in 2nd commit.